services/app-web/src/common-code/domain-models/submission.test.ts

```diff
@@ -359,4 +361,45 @@ describe('submission type assertions', () => {
 
         expect(submissionName(sub, programs)).toBe(expectedName)
     })
+
+    const rateNameTestArray: {
+        rateData: RateDataType
+        submissionName: string
+        expectedName: string
+    }[] = [
+        {
+            rateData: {
+                rateType: 'AMENDMENT',
+                rateDateStart: new Date('2021/05/21'),
+                rateDateEnd: new Date('2022/03/21'),
+                rateDateCertified: new Date('2021/04/23'),
+                rateAmendmentInfo: {
+                    effectiveDateStart: new Date('2022/05/21'),
+                    effectiveDateEnd: new Date('2022/09/21'),
+                },
+            },
+            submissionName: 'MN-TEST-NAME',
+            expectedName:
+                'MN-TEST-NAME-RATE-20220521-20220921-AMENDMENT-20210423',
+        },
+        {
+            rateData: {
+                rateType: 'NEW',
+                rateDateStart: new Date('2021/04/22'),
+                rateDateEnd: new Date('2022/03/29'),
+                rateDateCertified: new Date('2021/05/23'),
+            },
+            submissionName: 'OH-TEST-NAME',
+            expectedName:
+                'OH-TEST-NAME-RATE-20210422-20220329-CERTIFICATION-20210523',
+        },
+    ]
+    test.each(rateNameTestArray)(
+        'submission rate name is correct',
+        ({ rateData, submissionName, expectedName }) => {
+            expect(generateRateName(rateData, submissionName)).toMatch(
```

Comment:
I noticed that when testing the "table", if the first row passes, the entire test passes regardless if subsequent rows do not match. This happens with `submission name is correct`, change the last table row `expectedName` to something that will never match the function, the entire test will still pass.

Not sure whats going on. Any insights @macrael @haworku @rswerve ?

Reply:
Hm. locally I was able to change the last test in the list for “submission name is correct” and get an error.
<img width="702" alt="Screen Shot 2022-04-11 at 2 32 31 PM" src="https://user-images.githubusercontent.com/55759/162836990-9a7b1664-e519-456f-80b9-40a017d042d3.png">

Reply:
is there a reason you are using `toMatch()` instead of `toBe()` here?

Reply:
Oops, it was me the whole time! False alarm. I need to run these tests with `yarn test` and not using the built in Webstorm tool for jest.